src/Compilers/CSharp/Portable/Emitter/EditAndContinue/CSharpSymbolMatcher.cs

```diff
@@ -718,6 +728,9 @@ private bool AreNamedTypesEqual(NamedTypeSymbol type, NamedTypeSymbol other)
                 // TODO: Test with overloads (from PE base class?) that have modifiers.
                 Debug.Assert(!type.HasTypeArgumentsCustomModifiers);
                 Debug.Assert(!other.HasTypeArgumentsCustomModifiers);
+                Debug.Assert(!type.IsTupleType);
+                Debug.Assert(!other.IsTupleType);
```

Comment:
The only caller for `AreMethodsEqual` already had an assert that the method is a definition. Updated the code to check the type *parameters*, as that is the real intent.